dockerfile-redhat.template

```diff
@@ -0,0 +1,53 @@
+# Copyright 2022 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG ARCH
+ARG DISTRO_NAME
+ARG DISTRO_NICK
+
+FROM __DISTRO_REPO__
+
+ARG AT_VERSION
+ARG DISTRO_NAME
+ARG DISTRO_NICK
+ARG EXTRA
+ARG MINOR
+ARG REPO
+
+WORKDIR /root
+
+RUN dnf update \
+    && dnf install -y __BASE_PACKAGES__ \
+    && dnf clean all
+
+RUN curl -s --fail ${REPO}/${DISTRO_NAME}/${DISTRO_NICK^^}/__GPG_KEY__ -o __GPG_KEY__\
```

Comment:
I'm not sure how `DISTRO_NAME` and `DISTRO_NICK` are being populated (probably by hand?). If we end up using a copy of `lsb_release` like on the AT repo to automatically populate them, the Makefile would set `DISTRO_NICK` to:
```
   DISTRO_NICK := $(shell $(LSB_TOOL) -c -s | tr [:upper:] [:lower:])
```
```
$ advance-toolchain/scripts/utilities/lsb_release -s -c | tr [:upper:] [:lower:]
plow
```
But this would not generate the proper path for the GPG key.
Actually, `DISTRO_NICK` is only meaningful for Debian/Ubuntu. IMHO we should get rid of `DISTRO_NICK` for non-Debian based distros and use `DISTRO_RELEASE` instead:
```
$ scripts/utilities/lsb_release -s -r
9.0
```
Then this path could be written as `${REPO}/${DISTRO_NAME}/RHEL${DISTRO_RELEASE/.*}/__GPG_KEY__`